File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/FlightController/FCActionsTableViewController.m
//
//  FCActionsTableViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//

#import "FCActionsTableViewController.h"
#import "DemoSettingItem.h"
#import "DemoComponentHelper.h"
#import "FCLandingGearViewController.h"
#import "FCCompassViewController.h"
#import "FCOrientationViewController.h"
#import "FCVirtualStickViewController.h"
#import "FCFlightLimitationViewController.h"
#import "FCGeneralControlViewController.h"
#import "FCIntelligentAssistantViewController.h"
#import <DJISDK/DJISDK.h>

@interface FCActionsTableViewController ()

@end

@implementation FCActionsTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    DJIFlightController* fc = [DemoComponentHelper fetchFlightController];
    self.sectionNames = [NSMutableArray arrayWithArray:@[@"General", @"Orientation Mode", @"Virtural Stick"]];
    
    //General
    DemoSettingItem* item0 = [DemoSettingItem itemWithName:@"General Control" andClass:[FCGeneralControlViewController class]];
    DemoSettingItem* item1 = [DemoSettingItem itemWithName:@"Compass" andClass:[FCCompassViewController class]];
    DemoSettingItem* item2 = [DemoSettingItem itemWithName:@"Flight Limitation" andClass:[FCFlightLimitationViewController class]];
    DemoSettingItem* item3 = [DemoSettingItem itemWithName:@"Landing Gear" andClass:[FCLandingGearViewController class]];
    DemoSettingItem* item4 = [DemoSettingItem itemWithName:@"Intelligent Flight Assistant" andClass:[FCIntelligentAssistantViewController class]];
    NSMutableArray* array = [[NSMutableArray alloc] initWithArray:@[item0, item1, item2]];
    if (fc && fc.isLandingGearMovable) {
        [array addObject:item3];
    }
    if (fc && fc.intelligentFlightAssistant) {
        [array addObject:item4];
    }
    
    [self.items addObject:array];
    
    // Orientation Mode
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Orientation Mode" andClass:[FCOrientationViewController class]]]];
    
    // Virtual Stick
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Virtual Stick" andClass:[FCVirtualStickViewController class]]]];
}

-(DJIBaseComponent *)getComponent {
    return [DemoComponentHelper fetchFlightController]; 
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/FlightController/FCActionsTableViewController.swift
//
//  FCActionsTableViewController.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 16/1/4.
//  Copyright © 2016 DJI. All rights reserved.
//

import DJISDK

class FCActionsTableViewController: DemoTableViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Flight Controller"
        self.connectedComponent = ConnectedProductManager.sharedInstance.fetchFlightController()
        self.showComponentVersionSn = true
        
        self.configeItems()
    }
    
    func configeItems () {
    
        self.sectionNames = ["General", "Orientation Mode", "Virtural Stick", "Intelligent Assistant"]
        //General
        let item0:DemoSettingItem = DemoSettingItem(name: "General Control", andClass: FCGeneralControlViewController.self)
        let item1:DemoSettingItem = DemoSettingItem(name: "Compass", andClass: FCCompassViewController.self)
        let item2:DemoSettingItem = DemoSettingItem(name: "Flight Limitation", andClass: FCFlightLimitationViewController.self)
        
        let fc: DJIFlightController? = self.connectedComponent as? DJIFlightController
        
        if (fc != nil) {
            
            let  movable:Bool = fc!.isLandingGearMovable()
            
            if (movable == true) {
                let item3: DemoSettingItem = DemoSettingItem(name: "Landing Gear", andClass: FCLandingGearViewController.self)
                self.items.append([item0, item1, item2, item3])
            }else{
                self.items.append([item0, item1, item2])
            }
        }
        // Orientation Mode in storyboard
        self.items.append([DemoSettingItem(name:"Intelligent Orientation", andClass:nil)])
        // Virtual Stick in storyboard
        self.items.append([DemoSettingItem(name:"Virtual Stick", andClass:nil)])
        // Intelligent Assistent in storyboard
        if (fc != nil && fc?.intelligentFlightAssistant != nil){
            self.items.append([DemoSettingItem(name:"Intelligent Assistant", andClass:FCIntelligentAssistantViewController.self)])
        }
    }
    
    //Passes an instance of the current component selected to IndividualComponentViewController
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if (segue.identifier != "openComponentInfo") {
            let vc = segue.destinationViewController as! DJIBaseViewController
            vc.moduleTitle = segue.identifier
        }
    }
}